ibc-testkit/src/relayer/context.rs

```diff
@@ -1,97 +1,391 @@
+        port_id_on_b: PortId,
+        chan_id_on_a: ChannelId,
+        port_id_on_a: PortId,
+        signer: Signer,
+    ) {
+        let conn_id_on_b = self
+            .ctx_b
+            .ibc_store()
+            .channel_end(&ChannelEndPath::new(&port_id_on_b, &chan_id_on_b))
+            .expect("connection exists")
+            .connection_hops()[0]
+            .clone();
+
+        let conn_id_on_a = self
+            .ctx_a
+            .ibc_store()
+            .channel_end(&ChannelEndPath::new(&port_id_on_a, &chan_id_on_a))
+            .expect("connection exists")
+            .connection_hops()[0]
+            .clone();
+
+        let client_id_on_b = self
+            .ctx_b
+            .ibc_store()
+            .connection_end(&conn_id_on_b)
+            .expect("connection exists")
+            .client_id()
+            .clone();
+
+        let client_id_on_a = self
+            .ctx_a
+            .ibc_store()
+            .connection_end(&conn_id_on_a)
+            .expect("connection exists")
+            .client_id()
+            .clone();
+
+        TypedRelayerOps::<B, A>::close_channel_on_a(
+            &mut self.ctx_b,
+            &mut self.ctx_a,
+            client_id_on_b,
+            chan_id_on_b,
+            port_id_on_b,
+            client_id_on_a,
+            chan_id_on_a,
+            port_id_on_a,
+            signer,
+        )
+    }
+
+    /// Sends a packet from the first context to the second context.
+    /// The IBC packet is created by an IBC application on the first context.
+    pub fn send_packet_on_a(&mut self, packet: Packet, signer: Signer) {
+        let conn_id_on_a = self
+            .ctx_a
+            .ibc_store()
+            .channel_end(&ChannelEndPath::new(
+                &packet.port_id_on_a,
+                &packet.chan_id_on_a,
+            ))
+            .expect("connection exists")
+            .connection_hops()[0]
+            .clone();
+
+        let conn_id_on_b = self
+            .ctx_b
+            .ibc_store()
+            .channel_end(&ChannelEndPath::new(
+                &packet.port_id_on_b,
+                &packet.chan_id_on_b,
+            ))
+            .expect("connection exists")
+            .connection_hops()[0]
+            .clone();
+
+        let client_id_on_a = self
+            .ctx_a
+            .ibc_store()
+            .connection_end(&conn_id_on_a)
+            .expect("connection exists")
+            .client_id()
+            .clone();
+
+        let client_id_on_b = self
+            .ctx_b
+            .ibc_store()
+            .connection_end(&conn_id_on_b)
+            .expect("connection exists")
+            .client_id()
+            .clone();
+
+        TypedRelayerOps::<A, B>::send_packet_on_a(
+            &mut self.ctx_a,
+            &mut self.ctx_b,
+            packet,
+            client_id_on_a,
+            client_id_on_b,
+            signer,
+        )
     }
 }
 
 #[cfg(test)]
 mod tests {
```

Comment:
Is it okay to move these tests? It only has one test that checks if the client updates go well for four rounds. The new generic integration test function does multiple client updates already.

I am not in favor of removing this - any test is good test :slightly_smiling_face: Can we move this in `ibc-testkit/tests/relayer` directory?

Reply:
Makes sense. Perhaps it would be more appropriate to place it under the `ibc-testkit/tests/core/ics02_client/update_client.rs`. WDYT?

Reply:
66ad234